scheduler/generic_sched.go

```diff
@@ -591,6 +591,7 @@ func (s *GenericScheduler) computePlacements(destructive, place []placementResul
 
 			// Store the available nodes by datacenter
 			s.ctx.Metrics().NodesAvailable = byDC
+			s.ctx.Metrics().NodesInPool = len(nodes)
```

Comment:
That is odd. I suspect it's happening because each loop is adding other scoring metadata to the metrics, and we didn't think about where to set this. If the `s.ctx` changed I could see it, but that's set once for the whole eval.

Reply:
Ah nice find, thanks for digging! I'm glad we have good test coverage here 😄